client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/notification-settings/index.tsx

```diff
@@ -94,17 +97,36 @@ export default function NotificationSettings( {
 		'jetpack/pro-dashboard-monitor-sms-notification'
 	);
 
-	const mapAndStringifyEmails = ( emails: MonitorSettingsEmail[] ) => {
-		return JSON.stringify( emails.map( ( { name, email } ) => ( { name, email } ) ) );
+	const isContactListMatch = (
+		list1: ReadonlyArray< MonitorSettingsContact >,
+		list2: ReadonlyArray< MonitorSettingsContact >
+	): boolean => {
+		const stringifyContacts = ( list: ReadonlyArray< MonitorSettingsContact > ): string => {
+			return JSON.stringify(
+				list.map( ( item: MonitorSettingsContact ) => {
+					const property = (
+						'email' in item ? 'email' : 'phoneNumberFull'
+					) as keyof MonitorSettingsContact;
+					return {
+						name: item.name,
+						[ property ]: item[ property ],
+					};
+				} )
+			);
+		};
+
+		return stringifyContacts( list1 ) === stringifyContacts( list2 );
```

Comment:
It is possible that the user tries to delete a contact and then adds the same contact again. This will most likely update the order. Though we can ignore this scenario and just treat it as a new change.